### DISCUSSION TITLE
Help me help the community with the Azure hosting story

### DISCUSSION THREAD

**Discussion by dsghi** [OrchardCMS/OrchardCore#13608] [General]
What's the recommended Azure hosting story? We recently realized we had our App_Data folder committed to our Github repo; though it was private, it still feels like a huge risk (phishing, the day private repos mysteriously become public, etc.).  When we deleted it and purged it from our repo history, we broke the site (duh). We restored the directory via an app service backup, and now the repo is in a different state than the app service. (yikes)

There isn't coherent guidance around Azure hosting (this sounds rude, but it's constructive criticism). There are various modules to enable pieces of Azure; Media, Shells Configuration provider, and Keyvault. But they seem to have been developed separately. Guidance is spread between official docs, discussion topics, and PR comments.

The basics of utilizing any of these modules, or all of them, need clarification. Such as how to properly manage the settings variation between dev/test and production or for single vs. multi-tenant.

The challenge with docs is that they're often written by the dev-focused folks who create the modules, making them difficult for the layman to understand. If someone knows how these pieces work together for the optimal Azure hosting experience and is willing to give me a walk-through, I would happily translate that into documentation.

**Comment by hishamco** [comment]
I think Lombiq has a successor story with Azure Hosting, hope @Piedone or @BenedekFarkas can share some of their thoughts

**Reply by Piedone** [reply]
Indeed, thanks :).

**Comment by Piedone** [comment]
This'd be the basics of Azure hosting:

- [Configure SQL connection strings from the App Service slot's App Settings](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#global-tenant-data-access-configuration) (the example in the docs shows appsettings.json-based configs, but [there are many other options working similarly](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/)).
- Use the Azure Media Storage feature for media storage  [configure Azure Blob Storage connection strings](https://docs.orchardcore.net/en/latest/docs/reference/modules/Media.Azure/#configuration) as such.
- [Store shell settings in the DB](https://docs.orchardcore.net/en/latest/docs/reference/core/Shells/#database-shells-configuration-provider).
- [Store data protection key rings in Azure Blob Storage](https://docs.orchardcore.net/en/latest/docs/reference/modules/DataProtection.Azure/#purpose).
- [Use Elasticsearch](https://docs.orchardcore.net/en/latest/docs/reference/modules/Elasticsearch/) with an external indexing server instead of Lucene.
- +1 Use [Lombiq.Hosting.Azure.ApplicationInsights](https://github.com/Lombiq/Orchard-Azure-Application-Insights) to collect correlated (like server-side requests with client-side page views) telemetry. The rest of the [Lombiq Hosting Suite](https://dotnest.com/knowledge-base/topics/lombiq-hosting-suite) may be useful for you too.

For local development, you don't need an App_Data committed to the repository. Instead, store what's needed for a local development instance (content and configuration) in recipes, and run the setup with [Auto Setup](https://docs.orchardcore.net/en/latest/docs/reference/modules/AutoSetup/).

And indeed, this is not easy to figure out from the pieces, I'll see if I can provide a summary page too.

**Reply by dsghi** [reply]
This is great; thank you!
